android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -752,4 +840,310 @@ public static ErcToken createEthereumErcToken() {
+            for (String accountName : pendingTxInfos.keySet()) {
+                TransactionInfo[] txInfos = pendingTxInfos.get(accountName);
+                for (TransactionInfo txInfo : txInfos) {
+                    if (txInfo.txType == TransactionType.ERC20_TRANSFER
+                            || txInfo.txType == TransactionType.ERC20_APPROVE) {
+                        for (ErcToken token : tokens) {
+                            // Replace USDC and DAI contract addresses for Ropsten network
+                            token.contractAddress = getContractAddress(
+                                    chainId, token.symbol, token.contractAddress);
+                            String symbol = token.symbol;
+                            int decimals = token.decimals;
+                            if (txInfo.txType == TransactionType.ERC20_APPROVE) {
+                                symbol = "ETH";
+                                decimals = 18;
+                            }
+                            if (token.contractAddress.toLowerCase(Locale.getDefault())
+                                            .equals(txInfo.txData.baseData.to.toLowerCase(
+                                                    Locale.getDefault()))) {
+                                assets.put(txInfo.id, symbol);
+                                assetsDecimals.put(symbol, decimals);
+                                break;
+                            }
+                        }
+                    } else {
+                        assets.put(txInfo.id, "ETH");
+                        assetsDecimals.put("ETH", 18);
+                    }
+                }
+            }
+            fetchAssetsPrices(accountInfos, ethPrice, assetPrice, rvTransactions, callback, context,
+                    pendingTxInfos, assets, assetsDecimals, assetRatioController);
+        });
+    }
+
+    private static void fetchAssetsPrices(AccountInfo[] accountInfos, double ethPrice,
+            double assetPrice, RecyclerView rvTransactions, OnWalletListItemClick callback,
+            Context context, HashMap<String, TransactionInfo[]> pendingTxInfos,
+            HashMap<String, String> assets, HashMap<String, Integer> assetsDecimals,
+            AssetRatioController assetRatioController) {
+        AssetsPricesHelper assetsPricesHelper =
+                new AssetsPricesHelper(assetRatioController, assets);
+        assetsPricesHelper.fetchPrices(() -> {
+            workWithTransactions(accountInfos, ethPrice, assetPrice, rvTransactions, callback,
+                    context, "", 0, pendingTxInfos, assets, assetsDecimals,
+                    assetsPricesHelper.getAssetsPrices());
+        });
+    }
+
+    private static void workWithTransactions(AccountInfo[] accountInfos, double ethPrice,
+            double assetPrice, RecyclerView rvTransactions, OnWalletListItemClick callback,
+            Context context, String assetSymbol, int assetDecimals,
```

Comment:
how about to remove `Context context` argument and use instead `ContextUtils.getApplicationContext()`, just to have less parameters here and at `workWithTransactions`, `fetchAssetsPrices`, `fetchAssetsPricesDecimals`, `fetchTransactions`?

Reply:
I thought about that and made it for purpose to avoid extra unneeded calls. It's a question what is actually faster

Reply:
I thought about the readability more. So then lets keep it as is now.